src/com/xilinx/rapidwright/design/NetTools.java

```diff
@@ -46,4 +53,95 @@ public static boolean isGlobalClock(Net net) {
         
         return clkSrcSiteTypeEnums.contains(srcSpi.getSiteTypeEnum());
     }
+
+    public static class NodeTree extends Node {
```

Comment:
I think I can be persuaded to allow inner classes, but it seems like `NodeTree` has no relationship to `NetTools`, other than we are trying to hide it and `DesignTools` is too large.  Any reservations to making a regular class?

Reply:
Nevermind, I see there is a method at the bottom, I missed the indent.